x-pack/solutions/security/plugins/security_solution/public/detections/hooks/alert_summary/use_assistant.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useAssistantOverlay } from '@kbn/elastic-assistant';
+import { useCallback, useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import type { Alert } from '@kbn/alerting-types';
+import { flattenAlertType } from '../../utils/flatten_alert_type';
+import { getAlertFieldValueAsStringOrNull } from '../../utils/get_alert_field_value_as_string_or_null';
+import {
+  PROMPT_CONTEXT_ALERT_CATEGORY,
+  PROMPT_CONTEXTS,
+} from '../../../assistant/content/prompt_contexts';
+import {
+  ALERT_SUMMARY_CONTEXT_DESCRIPTION,
+  ALERT_SUMMARY_CONVERSATION_ID,
+  ALERT_SUMMARY_VIEW_CONTEXT_TOOLTIP,
+} from '../../../common/components/event_details/translations';
+
+const SUMMARY_VIEW = i18n.translate('xpack.securitySolution.eventDetails.summaryView', {
+  defaultMessage: 'summary',
+});
+
+export interface UseAssistantParams {
+  /**
+   * An array of field objects with category and value
+   */
+  alert: Alert;
+}
+
+export interface UseAssistantResult {
+  /**
+   * Function to show assistant overlay
+   */
+  showAssistantOverlay: (show: boolean) => void;
+}
+
+/**
+ * Hook to return the assistant button visibility and prompt context id.
+ * This is meant to be used in the AI for SOC tier, where the assistant is always enabled.
+ */
+export const useAssistant = ({ alert }: UseAssistantParams): UseAssistantResult => {
+  const getPromptContext = useCallback(async () => {
```

Comment:
```suggestion
  const getPromptContext = useCallback(() => {
```
It looks like nothing in this callback is asynchronous

Reply:
I will check again, but the `getPromptContext` is expecting a `Promise` so I think types were complaining. Also I naively copied this section from another place where we use a similarish `getPromptContext` callback, but it was used with `dataForFieldBrowser` which is an array of values, and in my case I have an `Alert` type

Reply:
Yup, the `async` is mandatory